src/stcal/jump/jump.py

```diff
@@ -262,3 +299,131 @@ def detect_jumps(frames_per_group, data, gdq, pdq, err,
 
     # Return the updated data quality arrays
     return gdq, pdq
+
+
+def flag_large_events(gdq, jump_flag, sat_flag, min_sat_area=1,
+                      min_jump_area=6,
+                      expand_factor=1.9, use_ellipses=False,
+                      sat_required_snowball=True):
+    """
+    This routine controls the creation of expanded regions that are flagged as jumps. These are called
+    snowballs for the NIR and are almost always circular with a saturated core. For MIRI they are better
+    fit with ellipses.
+
+    :param gdq: The group DQ cube for all integrations
+    :param jump_flag: The bit value that represents jump
+    :param sat_flag:  The bit value that represents saturation
+    :param min_sat_area:  The minimum area of saturated pixels within the jump circle to trigger the
+                        creation of a snowball.
+    :param min_jump_area:  The minimum area of a contour to cause the creation of the minimum
+                        enclosing ellipse or circle
+    :param expand_factor: The factor that increases the size of the snowball or enclosing ellipse.
+    :param use_ellipses:  Use ellipses rather than circles (better for MIRI)
+    :param sat_required_snowball: Require that there is a saturated pixel within the radius of the jump
+                        circle to trigger the formation of a snowball.
+    :return: none
+    """
+    for integration in range(gdq.shape[0]):
+        for group in range(1, gdq.shape[1]):
+            if use_ellipses:
+                jump_ellipses = find_ellipses(gdq[integration, group, :, :], jump_flag, min_jump_area)
+                log.info(f"Integration {integration} group {group} number of showers = {len(jump_ellipses)}.")
```

Comment:
This log message will contain 0-indexed values of integration and group. Just checking to make sure that's what you want, as opposed to 1-indexed.

Reply:
The first group doesn't have any jumps. The key reason it starts at index 1 is that I am using the change in the saturation flags to search for the core of the snowballs. That means I have to use the delta from the previous group. So, the loops needs to start at an index of 1.